### PR TITLE
Fixes #30552 - Fix vmware vm cloning when volumes are not present

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 1]
```diff
@@ -548,8 +548,8 @@ def clone_vm(raw_args)
         "numCPUs" => args[:cpus],
         "numCoresPerSocket" => args[:corespersocket],
         "memoryMB" => args[:memory_mb],
-        "datastore" => args[:volumes].first[:datastore],
-        "storage_pod" => args[:volumes].first[:storage_pod],
+        "datastore" => args[:volumes].empty? ? nil : args[:volumes].first[:datastore],
+        "storage_pod" => args[:volumes].empty? ? nil : args[:volumes].first[:storage_pod],
```
Comment (tbrisker) [inline app/models/compute_resources/foreman/model/vmware.rb:552]: This doesn't look like it's specific to hammer, would a clearer commit message be something like "fix vmware vm cloning when volumes are not present"?

Comment (yifatmakias) [reply app/models/compute_resources/foreman/model/vmware.rb:552]: Yes, I agree I will change the commit message.

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Review by shiramax — CHANGES_REQUESTED** [review]
@yifatmakias looks good!
can you add tests?

**Comment by yifatmakias** [comment]
@shiramax Added tests let me know if it is ok :)

**Comment by shiramax** [comment]
@yifatmakias code looks good, 
however, I'm not sure that the best way here is to pick the first volume. maybe if the volume didn't choose we should tell the user he must choose? 
Also, in the Redmine issue, I don't see that the bug is happening in VM clone but in host creation.

**Review by shiramax — APPROVED** [review]

**Comment by shiramax** [comment]
@yifatmakias we talked about this PR offline. good look good, merging this one.

**Comment by tbrisker** [comment]
2.3 - fbbc2975f

**Pull request merged**